avm/res/network/network-manager/main.bicep

```diff
@@ -240,3 +238,158 @@ type roleAssignmentType = {
   @sys.description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+type networkGroupType = {
+  @sys.description('Required. The name of the network group.')
+  name: string
+
+  @sys.description('Optional. A description of the network group.')
+  description: string?
+
+  @sys.description('Optional. Static Members to create for the network group. Contains virtual networks to add to the network group.')
+  staticMembers: {
+    @sys.description('Required. The name of the static member.')
+    name: string
+
+    @sys.description('Required. Resource ID of the virtual network.')
+    resourceId: string
+  }[]?
+}[]?
+
+type networkManagerScopeAccessType = ('Connectivity' | 'SecurityAdmin')[]
+
+type networkManagerScopesType = {
+  @sys.description('Optional. Management groups to assign to the network manager to manage.')
```

Comment:
I wonder if this would be a good case for `conditional` - in like 'Conditional. (...). Required if no Subscriptions were provided.')` 

On that note - Are these mangement group & subscriptions Ids? If so, I think we should either call this out - or even name the input accordingly to be explicit. I, for my part, "think" you mean the ID, but maybe it's the name? :D

Reply:
That's a good proposal. I would agree and align with the `Conditional` approach. These are resource IDs:

```json
            "managementGroups": [
                "/providers/Microsoft.Management/managementGroups/<mgId>"
            ],
            "subscriptions": [
                "/subscriptions/<subId>
            ],
```

I will call it out on the parameter to be more explicit.

Reply:
Can't see the update yet

Reply:
Updated